.editorconfig

```diff
@@ -0,0 +1,8 @@
+root = true
+
+# properties
+[*.{h,c,hpp,cpp,m,java}]
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
```

Comment:
I believe this config won't work with VS Code? (see internal discussion email)

Also, perhaps add note in the PR what editors you have tested with and which work and which doesn't, in terms of viewing and adding/editing code, so users know what to expect.

Reply:
This should work after the tab is expanded to spaces. It's not going to work if spaces is still use in the code.